src/requirements/Uniswap/UniswapForm.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Input,
+  Skeleton,
+  Stack,
+} from "@chakra-ui/react"
+import { consts } from "@guildxyz/types"
+import FormErrorMessage from "components/common/FormErrorMessage"
+import { useFormContext, useWatch } from "react-hook-form"
+import { RequirementFormProps } from "requirements"
+import ChainPicker from "requirements/common/ChainPicker"
+import MinMaxAmount from "requirements/common/MinMaxAmount"
+import TokenPicker from "requirements/common/TokenPicker"
+import parseFromObject from "utils/parseFromObject"
+import { Chains } from "wagmiConfig/chains"
+import { ADDRESS_REGEX, useParseVaultAddress } from "./hooks/useParseVaultAddress"
+import { useTokenSymbolsOfPoolVault } from "./hooks/useTokenSymbolsOfPoolVault"
+import { useEffect } from "react"
+
+const UniswapForm = ({
+  baseFieldPath,
+  field,
+}: RequirementFormProps): JSX.Element => {
+  const {
+    formState: { errors, touchedFields },
+    setValue,
+    clearErrors,
+    register,
+  } = useFormContext()
+  const isEditMode = !!field?.id
+
+  const chain: (typeof consts.UniswapV3PositionsChains)[number] = useWatch({
+    name: `${baseFieldPath}.chain`,
+  })
+
+  const lpVaultAddress = useParseVaultAddress(baseFieldPath)
+
+  const { error, isLoading, symbol0, symbol1, token0, token1 } =
+    useTokenSymbolsOfPoolVault(Chains[chain], lpVaultAddress)
+
+  useEffect(() => {
+    if (!token0 || !token1) {
+      return
+    }
+    setValue(`${baseFieldPath}.data.token0`, token0, { shouldDirty: true })
+    setValue(`${baseFieldPath}.data.token1`, token1, { shouldDirty: true })
+  }, [token0, token1])
+
+  // Reset form on chain change
+  const resetForm = () => {
+    if (!parseFromObject(touchedFields, baseFieldPath)?.address) return
+    setValue(`${baseFieldPath}.data.token0`, undefined)
+    setValue(`${baseFieldPath}.data.token1`, undefined)
+    setValue(`${baseFieldPath}.data.minAmount`, undefined)
+    setValue(`${baseFieldPath}.data.maxAmount`, undefined)
+    clearErrors([
+      `${baseFieldPath}.data.token0`,
+      `${baseFieldPath}.data.token1`,
+      `${baseFieldPath}.data.minAmount`,
+      `${baseFieldPath}.data.maxAmount`,
+    ])
+  }
+
+  return (
+    <Stack spacing={4} alignItems="start">
+      <ChainPicker
+        controlName={`${baseFieldPath}.chain` as const}
+        supportedChains={
+          consts.UniswapV3PositionsChains as unknown as Array<
+            (typeof consts.UniswapV3PositionsChains)[number]
+          >
+        }
+        onChange={resetForm}
+      />
+
+      {isEditMode ? (
+        <>
+          <TokenPicker
+            label={"Pair 1"}
```

Comment:
Should we use "Token 1" / "Token 2" for these fields? Since the user will pick one token in one input, not a pair?

Reply:
Makes sense! Changes in https://github.com/guildxyz/guild.xyz/pull/1158/commits/8b879837d2c7d5a5eb02b07a37e65a14b6339f3a 👍